Mathlib/RingTheory/HahnSeries/Basic.lean

```diff
@@ -524,4 +524,24 @@ theorem order_ofForallLtEqZero [Zero Γ] (f : Γ → R) (hf : f ≠ 0) (n : Γ)
 
 end LocallyFiniteLinearOrder
 
+section Truncate
+variable [Zero R]
+
+/-- Zeros out coefficients of a `HahnSeries` at indices equal to or greater than `c`. -/
+@[simps]
+def truncate [PartialOrder Γ] (c : Γ) (x : HahnSeries Γ R) : HahnSeries Γ R where
+  coeff i := open Classical in if c ≤ i then 0 else x.coeff i
+  isPWO_support' := Set.IsPWO.mono x.isPWO_support (by simp)
```

Comment:
I am a bit nervous about `open Classical in` in a definition (this is not a proof). If Gamma is something like the integers then there is a constructive algorithm for `c ≤ i` and this approach will I think override it and use LEM which might lead to diamonds later? I think it might be better to assume `[DecidableLE Γ]` in the definition.